src/providers/MeetingProvider/MeetingManager.ts

```diff
@@ -671,10 +664,25 @@ export class MeetingManager implements AudioVideoObserver {
         this.selectVideoInputDeviceError = error;
         console.error('Failed to choose video input device.', error);
       }
-      this.selectedVideoInputDevice = deviceId;
+      let innerDevice = null;
+      if (isVideoTransformDevice(device)) {
+        innerDevice = await device.intrinsicDevice();
+      } else {
+        innerDevice = device;
+      }
+
+      const deviceId =
+        DefaultDeviceController.getIntrinsicDeviceId(innerDevice);
+      if (typeof deviceId === 'string') {
+        this.selectedVideoInputDevice = deviceId;
+      } else if (Array.isArray(deviceId) && deviceId[0]) {
+        this.selectedAudioInputDevice = deviceId[0];
```

Comment:
Should this be `this.selectedVideoInputDevice` ?

Reply:
oh gosh thanks